configuration/azure/appconfig/appconfig.go

```diff
@@ -0,0 +1,195 @@
+	s := &ConfigurationStore{
+		json:   jsoniter.ConfigFastest,
+		logger: logger,
+	}
+
+	return s
+}
+
+// Init does metadata and connection parsing.
+func (r *ConfigurationStore) Init(metadata configuration.Metadata) error {
+	m, err := parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	r.metadata = m
+
+	coreClientOpts := azcore.ClientOptions{
+		Telemetry: policy.TelemetryOptions{
+			ApplicationID: "dapr-" + logger.DaprVersion,
+		},
+	}
+
+	options := azappconfig.ClientOptions{
+		ClientOptions: coreClientOpts,
+	}
+
+	if r.metadata.connectionString != "" {
+		r.client, err = azappconfig.NewClientFromConnectionString(r.metadata.connectionString, &options)
+		if err != nil {
+			return err
+		}
+	} else {
+		var settings azauth.EnvironmentSettings
+		settings, err = azauth.NewEnvironmentSettings("appconfig", metadata.Properties)
+		if err != nil {
+			return err
+		}
+
+		var cred azcore.TokenCredential
+		cred, err = settings.GetTokenCredential()
+		if err != nil {
+			return err
+		}
+
+		r.client, err = azappconfig.NewClient(r.metadata.host, cred, &options)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func parseMetadata(meta configuration.Metadata) (metadata, error) {
+	m := metadata{}
+
+	if val, ok := meta.Properties[host]; ok && val != "" {
+		m.host = val
+	}
+
+	if val, ok := meta.Properties[connectionString]; ok && val != "" {
+		m.connectionString = val
+	}
+
+	if m.connectionString != "" && m.host != "" {
+		return m, fmt.Errorf("azure store error: can't set both %s and %s fields in metadata", host, connectionString)
+	}
+
+	if m.connectionString == "" && m.host == "" {
+		return m, fmt.Errorf("azure store error: specify %s or %s field in metadata", host, connectionString)
+	}
+
+	m.maxRetries = defaultMaxRetries
+	if val, ok := meta.Properties[maxRetries]; ok && val != "" {
+		parsedVal, err := strconv.Atoi(val)
+		if err != nil {
+			return m, fmt.Errorf("azure store error: can't parse maxRetries field: %s", err)
+		}
+		m.maxRetries = parsedVal
+	}
+
+	m.maxRetryDelay = defaultMaxRetryDelay
+	if val, ok := meta.Properties[maxRetryDelay]; ok && val != "" {
+		parsedVal, err := strconv.Atoi(val)
+		if err != nil {
+			return m, fmt.Errorf("azure store error: can't parse maxRetryDelay field: %s", err)
+		}
+		m.maxRetryDelay = time.Duration(parsedVal)
+	}
+
+	m.retryDelay = defaultRetryDelay
+	if val, ok := meta.Properties[retryDelay]; ok && val != "" {
+		parsedVal, err := strconv.Atoi(val)
+		if err != nil {
+			return m, fmt.Errorf("azure store error: can't parse retryDelay field: %s", err)
+		}
+		m.retryDelay = time.Duration(parsedVal)
+	}
+
+	return m, nil
+}
+
+func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
+	keys := req.Keys
+	items := make([]*configuration.Item, 0, len(keys))
+	for _, key := range keys {
+		resp, err := r.client.GetSetting(
+			ctx,
+			key,
```

Comment:
There is support for querying multiple keys 5 of them at a time in the service... https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-key-value#query-key-values

Is that something that needs to be looked at for this request?